File: src/main/java/com/ject/studytrip/global/batch/scheduler/BatchJobScheduler.java
package com.ject.studytrip.global.batch.scheduler;

import lombok.RequiredArgsConstructor;
import org.springframework.batch.core.Job;
import org.springframework.batch.core.JobParameters;
import org.springframework.batch.core.JobParametersBuilder;
import org.springframework.batch.core.launch.JobLauncher;
import org.springframework.context.annotation.Profile;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Profile("!test")
@Component
@RequiredArgsConstructor
public class BatchJobScheduler {
    private final JobLauncher jobLauncher;
    private final Job hardDeleteJob;

    // 매일 04:00 (Asia/Seoul)
    @Scheduled(cron = "0 0 4 * * *", zone = "Asia/Seoul")
    public void runHardDeleteJob() throws Exception {
        JobParameters jobParameters =
                new JobParametersBuilder()
                        .addLong("ts", System.currentTimeMillis())
                        .toJobParameters();

        jobLauncher.run(hardDeleteJob, jobParameters);
    }
}

File: src/main/kotlin/com/ject/studytrip/global/batch/scheduler/BatchJobScheduler.kt
package com.ject.studytrip.global.batch.scheduler

import org.springframework.batch.core.Job
import org.springframework.batch.core.JobParameters
import org.springframework.batch.core.JobParametersBuilder
import org.springframework.batch.core.launch.JobLauncher
import org.springframework.context.annotation.Profile
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Profile("!test")
@Component
class BatchJobScheduler(
    private val jobLauncher: JobLauncher,
    private val hardDeleteJob: Job,
) {
    companion object {
        private const val JOB_PARAMETER_TIMESTAMP = "ts"
    }

    /**
     * 매일 04:00 (Asia/Seoul)
     */
    @Scheduled(cron = "0 0 4 * * *", zone = "Asia/Seoul")
    fun runHardDeleteJob() {
        jobLauncher.run(hardDeleteJob, createJobParameters())
    }

    private fun createJobParameters(): JobParameters =
        JobParametersBuilder()
            .addLong(JOB_PARAMETER_TIMESTAMP, System.currentTimeMillis())
            .toJobParameters()
}
